File: CSharp/CountryApi/CountryApi/Models/Country.cs
using System.Collections.Generic;

namespace CountryApi.Models
{
    public class Country
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Code { get; set; }
        public List<FunFact> FunFacts { get; set; }
        public Country()
        {
            Id = default;
            Name = default;
            Code = default;
            FunFacts = new List<FunFact>();
        }
        public Country(string name, string code, List<FunFact> funFacts = null)
        {
            Id = default;
            Name = name;
            Code = code;
            FunFacts = funFacts ?? new List<FunFact>();
        }
        public Country(Country country)
        {
            Id = country.Id;
            Name = country.Name;
            Code = country.Code;
            FunFacts = country.FunFacts;
        }
    }
}

File: Java/country-api/src/main/java/com/countryapi/resources/Country.java
package com.countryapi.resources;

import java.util.ArrayList;

public class Country {
    private final int id;
    private final String name;
    private final String code;
    private final ArrayList<FunFact> funFacts;

    public Country(){
        this.id = 0;
        this.name = "";
        this.code = "";
        this.funFacts = new ArrayList<>();
    }
    public Country(int id, String name, String code, ArrayList<FunFact> funFacts) {
        this.id = id;
        this.name = name;
        this.code = code;
        this.funFacts = funFacts;
    }
    public Country(Country country){
        this.id = country.id;
        this.name = country.name;
        this.code = country.code;
        this.funFacts = country.funFacts;
    }

    private int getId(){
        return id;
    }
    private String getName(){
        return name;
    }
    private String getCode(){
        return code;
    }
    private ArrayList<FunFact> getFunFacts(){
        return funFacts;
    }
}
